### ISSUE TITLE
2 ERRORS in make&take

### ISSUE THREAD

**Issue by WalterL-wL** [thomas-maeder/popeye#341]
During extended checks with Circe Variants in PopEye v4.87 I stumbled across two errors under conditions "bedi Make&TakeSchach EquipollentsCirce" (h#, abbreviations in German notation, T=Rook, L=Bishop, B=Pawn). PopEye presented the following solution: 1.Kc7×Lh3 Tb8×d7[+Bf6]...

1. First, the capture move Tb8×d7 is ILLEGAL under make&take! According to its inventor H. Laue "... No pawn can be captured by a piece from the 1st or 8th rank"!

2. The second one [+Bf6] is equivalent to the EquipollentsCirce issue in #340 (see there): the vector to the rebirth square is wrongly derived from the composite move of make&take, it should only be derived from the take-move, so actually it's [+Bf7] (but still illegal!)..., because in EquipollentsCirce the VECTOR is defined by departure square of "capturer" to capture square (Märchenschachlexikon: "...wenn man den Zugvektor des *schlagenden Steins* an das Schlagfeld legt"; BCPS: "the length and direction of the *capture move* is the same as the length and direction from the capture square to the rebirth square"...

**Comment by thomas-maeder** [comment]
Please the complete input.

**Comment by WalterL-wL** [comment]
see attached the 'error-log' (German notation, my remarks added) from a test session (triggered by JF-1687), showing ILLEGAL black Pawn captures by (m&t-compound)moves from row 8 (the same would be row 1 for a white Pawn) **1... Tb8×d7[+sBf6]**. Further it seems to prove that Popeye adds the **whole** compound move (both steps in m&t, the same in t&m) as a vector to the capture square (under Equipollents- and in (Kontra)ParrainCirce).

H.Laue in f-235 and Schwalbe 298-1 (m&t): *"A pawn cannot be placed on its 1st or 8th rank either in the diagram or as the result of the make part of a make&take capture. No pawn can be captured by a piece from the 1st or 8th rank."*

Interesting also H.Laue in Schwalbe 229 (t&m): *"2) ...pawns can never move to (or initially be placed on) the first rank of their own colour, pawn promotion takes place only if the pawn move ends on the 8th rank (and not immediately after the *'take' part of a take&make move)"*. Further *"3) In the combination take&make-Circe, a Circe rebirth takes place only after the complete take&make move, not immediately after its 'take' part"*. But this is about the capture (take-move), not necessarily about the **whole** compound move! H. Laue made that immediately clear in the footnote on page 368 [my translation]: *"One could also imagine some kind of combination like 'Circe-take&make'... when rebirth after a capture (='take'-part) obtains priority over the 'make'-part of the move..."*. This specific alternative was also confirmed by him subsequently in f-223! So, this would actually **exclude** the 'make'-part of the compound move from the vector! To apply the whole 2-part compound move (in m&t as well as in t&m) as a vector to the capture square seems to be an **additional** option only, but beyond the regularly intended treatment by H.Laue...

THANKS!

[error-log CIRCE-vector + m&t (triggered by JF-1687).txt](https://github.com/thomas-maeder/popeye/files/9372981/error-log.CIRCE-vector.%2B.m.t.triggered.by.JF-1687.txt)

**Comment by thomas-maeder** [comment]
Let's deal with your issues one by one:

1. you write "the capture move Tb8×d7 is ILLEGAL under make&take"
I agree, but your example is Make&Take plus Circe Equipollent. In Circe Equipollent, pawns are allowed single steps from their side's base line, so a piece capturing a pawn from a square on it's side's base line must be allowed to do a single step in the Make part.
I am sure that Hartmut didn't think of Circe Equipollent when he made his statement.

**Comment by thomas-maeder** [comment]
2. you write "the vector to the rebirth square is wrongly derived from the composite move of make&take, it should only be derived from the take-move"

First a side note: I think that "take-move" is an unlucky term. The move consists of a Make and a Take part, so "take part" would be more precise.

Second, to your actual point: you express your view here which differs from what Popeye currently does. Popeye in its current implementation behaves according to my intuitive understanding of how the combination of the two conditions should work. I see no law or logical reason why one view should be prefered to the other.

**Comment by WalterL-wL** [comment]
> Let's deal with your issues one by one:
> 1. you write "the capture move Tb8×d7 is ILLEGAL under make&take"
> I agree, but your example is Make&Take plus Circe Equipollent. In Circe Equipollent, pawns are allowed single steps from their side's base line...

**good point!** I assume this Pawn move situation from own's side base line has been defined for Circe Equipollent already (couldn't find it), it definitely has been for Circe Parrain. Because with PWC or AntisuperC a Pawn on his side's base line would not be allowed to move at all (according to Märchenschachlexikon der Schwalbe)! Has this anything to do with the special Popeye condition "NormalPawn" in this respect?

H.Laue might not have thought of Circe Equipollent when he made his statement, but I once had a written discussion with him concerning Pawns on their side's base line (as basis for my article "Auffälligkeiten" in f-242, see especially 3.2 Fairy-Pawns), his opinion was totally clear then: There is his **absolute ban** on Pawns on their side's base line with the proper m&t condition, no exceptions - thus, no pawn can be captured by a piece from the 1st or 8th rank. Or should we generally follow the idea of A.Beine in Schwalbe 304 with problem #18298 (Annanchess+m&t: wSg1; 1.g2-g6×bRh8=wQ...; "...if it's an allowed move in one component of a pack of conditions then it's also allowed for the combination"... maybe it should rather be an extra option with Circe Equipollent?

**Comment by WalterL-wL** [comment]
> 2. you write "the vector to the rebirth square is wrongly derived from the composite move of make&take, it should only be derived from the take-move"

Agreed there's generally no law in the fairy field why one view should be preferred to the other. But there are definitions for Circe Equipollent (valid for m&t as well as t&m): the VECTOR is defined by departure square of **"capturer"** to capture square (Märchenschachlexikon der Schwalbe: "*wenn man den Zugvektor des **schlagenden** Steins an das Schlagfeld legt*"; BCPS: "*the length and direction of the **capture move** is the same as the length and direction from the capture square to the rebirth square*"...). So, the vector seems to be defined for the capture-act only (= take-part of the compound move), it would not be defined for the compound move as a whole.

And also there are some logical reasons against the current implementation of the vector assembled from the entire compound move. How would you perform H.Laue's own alternate ideas then? In (1) **Circe-t&m**: the rebirth act must be established following the take-part BEFORE the make-part is performed (in order to apply the whole compound move you wouldn't have done the make-part yet!); the whole compound move is illegal if the make-part could not be performed due to a single only available field already occupied from the rebirth act! In (2) **t&m-Circe**: the rebirth act from the take-part is performed AFTER the make-part has been done; there's no problem if the rebirth cannot take place due to a rebirth field that's already occupied by the make-part (which would be impossible with a vector from the compound move under Circe Equipollents!). There would always be a different rebirth field for the captured stone depending on the composition of the vector (take-part only, or compound move) to be applied to the capture field...

In m&t of course it's not about the priority of the rebirth act, but only about the composition of the vector and the resulting rebirth field, which can better be argued with t&m, but in the same sense insofar. And it's not just about rebirth after this or that part of the move, there are other conditions that could apply at different times of the two-part move (or even with make&take&make of the three-part move), like e.g. paralysis (Madrasi...), or change of color (Andernach, Masand/magic...), or observation (Isardam, Patrouille, Heffalumps...), ... etc. It's actually a matter of **priorities** (see feenschach 80. TT "Priority") and not a predetermined and one-way issue...

Maybe an interpretation of the vector representing the whole compound move could be offered as an additional option, resulting in different outcomes?

The situation with the vector in Circe Parrain seems completely different insofar, as the definition does not refer to the capture-act (= take-part) itself, but to the vector of the "following move of a stone". Insofar you can feasibly derive the vector even from a following compound move (m&t / t&m) of a stone. (The respective remarks in my error-log with Circe ContraParrain are obsolete insofar).

**Comment by WalterL-wL** [comment]
> And also there are some logical reasons against the current implementation of the vector assembled from the entire compound move.

I feel my last post needs some clarification concerning "some logical reasons": In any take&make with Circe I did not want to challenge Popeye's "fixed" priority handling (as: 1.take, 2.make and only then 3.rebirth), anyhow make&take with Circe does not allow a variation at all (it's always: 1.make, 2.take, 3.rebirth). But I tried to use this reference in order to show that the alternative priority handling like in Circe-take&make (1.take, 2.rebirth, 3.make) would technically **not even allow** to include the make-part into the rebirth vector of Circe Equipollents - because it hasn't yet happened in the moment of rebirth! *And as the compilation of the vector should be independent of this kind of priority interpretation(!!!), the make-part should not be part of the vector in Circe Equipollents at all, neither in make&take with Circe Equipollents!*
